core/src/main/scala/com/evolutiongaming/catshelper/FromFuture.scala

```diff
@@ -14,12 +14,25 @@ trait FromFuture[F[_]] {
 
 object FromFuture {
 
+  /** Convinient summon-and-run function
+   *
+   * @example {{{
+   *          class Foo[F[_]: FromFuture: Monad](db: FutureDB){
+   *            def query(lines: Int): F[Unit] = FromFuture.defer[F](db.query(lines))
+   *          }
+   * }}} */
+  def defer[F[_]]: FromFutureDeferPA[F] = new FromFutureDeferPA[F]()
+
+  class FromFutureDeferPA[F[_]](val __ : Boolean = true) extends AnyVal {
```

Comment:
private ?

Reply:
I'm afraid it's not possible as `class FromFutureDeferPA escapes its defining scope blah blah...`

Reply:
try `private[FromFuture]`

Reply:
This would mean that it would be only accessible from object `FromFuture` I believe.
But when user writes `val d = FromFuture.defer[F]`, the type of `d` will be `FromFutureDeferPA`,and that won't work.

Reply:
just try :)

Reply:
Wow, that's something new for me, thanks.